registry/txt.go

```diff
@@ -224,18 +248,31 @@ func (im *TXTRegistry) generateTXTRecord(r *endpoint.Endpoint) []*endpoint.Endpo
 			endpoints = append(endpoints, txt)
 		}
 	}
-	// new TXT record format (containing record type)
+
 	recordType := r.RecordType
 	// AWS Alias records are encoded as type "cname"
 	if isAlias, found := r.GetProviderSpecificProperty("alias"); found && isAlias == "true" && recordType == endpoint.RecordTypeA {
 		recordType = endpoint.RecordTypeCNAME
 	}
-	txtNew := endpoint.NewEndpoint(im.mapper.toNewTXTName(r.DNSName, recordType), endpoint.RecordTypeTXT, r.Labels.Serialize(true, im.txtEncryptEnabled, im.txtEncryptAESKey))
-	if txtNew != nil {
-		txtNew.WithSetIdentifier(r.SetIdentifier)
-		txtNew.Labels[endpoint.OwnedRecordLabelKey] = r.DNSName
-		txtNew.ProviderSpecific = r.ProviderSpecific
-		endpoints = append(endpoints, txtNew)
+
+	// "new" (version 2) TXT record format (containing record type)
+	if im.txtFormat == TXTFormatTransition && (recordType == endpoint.RecordTypeAAAA || recordType == endpoint.RecordTypeCNAME) {
+		txtNew := endpoint.NewEndpoint(im.mapper.toNewTXTName(r.DNSName, recordType), endpoint.RecordTypeTXT, r.Labels.Serialize(true, im.txtEncryptEnabled, im.txtEncryptAESKey))
+		if txtNew != nil {
+			txtNew.WithSetIdentifier(r.SetIdentifier)
+			txtNew.Labels[endpoint.OwnedRecordLabelKey] = r.DNSName
+			txtNew.ProviderSpecific = r.ProviderSpecific
+			endpoints = append(endpoints, txtNew)
+		}
+	}
+
+	// Metadata TXT record format
```

Comment:
which version ? old/new?
The comment I can get from the variable name `txtMetadata`.

Reply:
As `old` and `new` age badly, the name `metadata` for this format was introduced (see the PR and issues referenced above). This is the format that is supposed to replace the other 2 at some point in time, meaning that the `old` and `new` should be removed.

Reply:
Then I would just drop the comment.